src/lib/storage/yaml_writer.rb

```diff
@@ -119,30 +107,52 @@ def yaml_disk(disk)
         { "disk" => content }
       end
 
+      # Returns a YAML representation of the partitions and free slots in a disk
+      #
+      # @param disk [::Storage::Disk]
+      # return [Array<Hash>]
+      def yaml_disk_partitions(disk)
+        cyl_size = DiskSize.new(disk.geometry.cylinder_size / 1024)
+        first_free_cyl = 0
+        partitions = []
+        disk.partition_table.partitions.each do |partition|
+          gap = partition.region.start - first_free_cyl
+          partitions << yaml_free_slot(cyl_size * gap) if gap > 0
+
+          partitions << yaml_partition(partition)
+          if partition.type != ::Storage::PartitionType_EXTENDED
+            first_free_cyl = partition.region.end + 1
+          end
+        end
+        partitions
+      end
+
       #
       # Return the YAML counterpart of a ::Storage::Partition.
       #
       # @param  partition [::Storage::Partition]
       # @return [Hash]
       #
+      # rubocop:disable Metrics/AbcSize
       def yaml_partition(partition)
         content = {}
         content["size"] = DiskSize.new(partition.region.to_kb(partition.region.length)).to_s
         content["name"] = partition.name
         content["type"] = @partition_types[partition.type]
-        content["id"  ] = @partition_ids[partition.id]
+        content["id"] = @partition_ids[partition.id]
```

Comment:
well, this break intention of aligned assignment, but it also reveals fact, that there is not reason to not direct add it to hash like:
```
content = {
  "size"  => DiskSize.new(partition.region.to_kb(partition.region.length)).to_s,
  "name => partition.name,
  ...
}
```
which itself is proper aligned and it is ensured by rubocop